.ddev/config.yaml

```diff
@@ -0,0 +1,35 @@
+name: astrosite
+type: php
+docroot: dist
+webserver_type: nginx-fpm
+additional_hostnames: []
+additional_fqdns: []
+performance_mode: none
+use_dns_when_possible: true
+web_environment: []
+nodejs_version: "18"
+
+# 4321 is Astro Vite default
+web_extra_exposed_ports:
+- name: astro-dev
+  container_port: 4321
+  http_port: 4322
+  https_port: 4321
+
+# The extra -- is a Vite item
+# https://github.com/vitejs/vite/discussions/3396
+web_extra_daemons:
+  - name: astro-dev-deamon
+    command: "npm run dev -- --host"
+    directory: /var/www/html
+hooks:
+  post-start:
+    - exec: echo '================================================================================='
+    - exec: echo '                                  NOTICE'
+    - exec: echo '================================================================================='
+    - exec: echo 'The Astro dev container is ready.'
+    - exec: echo
+    - exec: echo "By default it will be available at port 4321 - URL:4321"
+    - exec: echo "Use ddev describe to confirm if this doesn't work."
+    - exec: echo 'Check the status of startup by running "ddev logs --follow --time"'
+    - exec: echo '================================================================================='
```

Comment:
* It will be a lot easier to make these one single exec, use newlines. 
* It might make more sense to use `exec-host` for this as well, since there's no need that I know of to execute inside the container.